File: Els_kom_Launcher_Stub/Forms/Form1.vb
Option Strict Off
Option Explicit On
Friend Class Form1
	Inherits Form

	Private ElsDir As String
	Private Sub Form1_Load(ByVal eventSender As Object, ByVal eventArgs As EventArgs) Handles MyBase.Load
		If IO.File.Exists(My.Application.Info.DirectoryPath & "\Settings.ini") Then
			ElsDir = Els_kom_Core.Classes.INIread.ReadIniValue(My.Application.Info.DirectoryPath & "\Settings.ini", "Settings.ini", "ElsDir")
			If ElsDir.Length > 0 Then
				If IO.File.Exists(ElsDir & "\voidels.exe") Then
					Els_kom_Core.Classes.Process.Shell(ElsDir & "\voidels.exe", "", False, False, False, AppWinStyle.NormalFocus, ElsDir)
					Me.Close()
				Else
					If IO.File.Exists(ElsDir & "\elsword.exe") Then
						Els_kom_Core.Classes.Process.Shell(ElsDir & "\elsword.exe", "", False, False, False, AppWinStyle.NormalFocus, ElsDir)
						Me.Close()
					Else
						MsgBox("Can't find '" & ElsDir & "\elsword.exe'. Make sure the File Exists and try to update Elsword Again!", MsgBoxStyle.Critical, "Error!")
						Me.Close()
					End If
				End If
			Else
				MsgBox("The Elsword Directory Setting is not set. Make sure to Set your Elsword Directory Setting and try to update Elsword Again!", MsgBoxStyle.Critical, "Error!")
				Me.Close()
			End If
		Else
			MsgBox("The Elsword Directory Setting is not set. Make sure to Set your Elsword Directory Setting and try to update Elsword Again!", MsgBoxStyle.Critical, "Error!")
			Me.Close()
		End If
	End Sub
End Class

File: Els_kom_Launcher_Stub/Forms/Form1.cs
using System.Windows.Forms;

namespace Els_kom_Launcher_Stub
{
	public partial class Form1 : Form
	{
		public Form1()
		{
			InitializeComponent();
		}
		
		private string ElsDir;
		
		private void Form1_Load(object sender, System.EventArgs e)
		{
			if (System.IO.File.Exists(Application.StartupPath + "\\Settings.ini"))
			{
				Els_kom_Core.Classes.INIObject settingsini = new Els_kom_Core.Classes.INIObject(Application.StartupPath + "\\Settings.ini");
				ElsDir = settingsini.Read("Settings.ini", "ElsDir");
				if (ElsDir.Length > 0)
				{
					if (System.IO.File.Exists(ElsDir + "\\voidels.exe"))
					{
						Els_kom_Core.Classes.Process.Shell(ElsDir + "\\voidels.exe", "", false, false, false, System.Diagnostics.ProcessWindowStyle.Normal, ElsDir, true);
						this.Close();
					}
					else
					{
						if (System.IO.File.Exists(ElsDir + "\\elsword.exe"))
						{
							Els_kom_Core.Classes.Process.Shell(ElsDir + "\\elsword.exe", "", false, false, false, System.Diagnostics.ProcessWindowStyle.Normal, ElsDir, true);
							this.Close();
						}
						else
						{
							MessageBox.Show("Can't find '" + ElsDir + "\\elsword.exe'. Make sure the File Exists and try to update Elsword Again!", "Error!", MessageBoxButtons.OK, MessageBoxIcon.Error);
							this.Close();
						}
					}
				}
				else
				{
					MessageBox.Show("The Elsword Directory Setting is not set. Make sure to Set your Elsword Directory Setting and try to update Elsword Again!", "Error!", MessageBoxButtons.OK, MessageBoxIcon.Error);
					this.Close();
				}
			}
			else
			{
				MessageBox.Show("The Elsword Directory Setting is not set. Make sure to Set your Elsword Directory Setting and try to update Elsword Again!", "Error!", MessageBoxButtons.OK, MessageBoxIcon.Error);
				this.Close();
			}
		}
	}
}
